### ISSUE TITLE
AAE Fullsync Incompatible with Bucket Types [JIRA: RIAK-1655]

### ISSUE THREAD

**Issue by ian-mi** [basho/riak_repl#663]
When using AAE fullsync, the fullsync source will crash when attempting to send a key with a bucket type with an error such as

```
2015-03-30 13:47:10.371 [error] <0.2216.0> Supervisor riak_repl2_fssource_sup had child 433883298582611803841718934712646521460354973696 started with {riak_repl2_fssou
rce,start_link,undefined} at <0.2163.1> exit with reason {{function_clause,[{riak_repl_util,new_w1,[{<<"type">>,<<"test">>},<<"test">>,<<53,1,0,0,0,38,131,108,0,0,0,1,
104,2,109,0,0,0,12,197,82,177,11,248,33,144,157,0,0,78,34,104,2,97,1,110,5,0,16,49,142,206,14,106,0,0,0,1,0,0,0,6,1,118,97,108,117,101,0,0,0,174,0,0,5,147,0,11,106,80,
0,9,163,133,22,55,75,103,66,118,54,102,66,121,102,103,116,104,54,84,80,68,98,56,77,79,70,0,0,0,0,12,1,88,45,82,105,97,107,45,77,101,116,97,0,0,0,3,0,131,106,0,0,0,6,1,
105,110,100,101,120,0,0,0,3,0,131,106,0,0,0,13,1,...>>],...},...]},...} in context child_terminated
```

It appears that this is happening because fullsync has not been updated to use the new replication wire protocol.

**Comment by ian-mi** [comment]
This issue does not occur with keylist fullsync, because `riak_repl_util:encode_obj_msg` is called with the `w0` wire protocol version. In AAE fullsync it is being called with `w2` which is not being handled correctly.

**Comment by Basho-JIRA** [comment]
There is a fix for this via [~bsparrow435]. We should schedule it for 2.0.6 and 2.1.1. 

**_[posted via JIRA by Sean Cribbs]**_

**Comment by Basho-JIRA** [comment]
https://github.com/basho/riak_repl/pull/692 reverts the original fix - if we decide to move forward with it for 2.1.2, it's here for reference. 

**_[posted via JIRA by Douglas Rohrer]**_

**Comment by Basho-JIRA** [comment]
Also, duplicate of riak-1987 

**_[posted via JIRA by Douglas Rohrer]**_

**Comment by Basho-JIRA** [comment]
Ian is working on the binary vs term object as it dances through the realtime processes.

PR is here https://github.com/basho/riak_repl/pull/682

Once completed will need to be applied against 2.0 (for 2.0.7) and 2.1 (for 2.1.2) - if possible target 2.0 and forward merge. 

**_[posted via JIRA by Jon Meredith]**_

**Comment by Basho-JIRA** [comment]
Merged PR onto 2.0 branch.
Merge forward is underwary, will set fix versions once bors merges. 

**_[posted via JIRA by Jon Meredith]**_
